### PR TITLE
Merge sort implemented. Test cases implemented.

### PR DESCRIPTION
**Pull request by georgikavalov**
Task three of part one is completed.

### CODE REVIEW

File: src/main/java/com/musala/edu/introduction/three/TaskThree.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+public class TaskThree {
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(TaskThree.class);
+	private static int[] tempArray;
+
+	/**
+	 * Constructor
+	 */
+	private TaskThree() {
+		throw new IllegalStateException("TaskThree is a static class");
```
Comment (ivorusev) [inline src/main/java/com/musala/edu/introduction/three/TaskThree.java]: No need for this. The constructor can be called only within the class anyways. Just leave a comment why it is empty, something like:

> // Hiding the default constructor for util class.

File: src/main/java/com/musala/edu/introduction/three/TaskThree.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@code TaskThree} is a static class
+ * 
+ * @author georgi.kavalov
+ *
+ */
+public class TaskThree {
```
Comment (ivorusev) [inline src/main/java/com/musala/edu/introduction/three/TaskThree.java]: And what was task three? Wouldn't it be better if it was some more meaningful name?

File: src/main/java/com/musala/edu/introduction/three/TaskThree.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+	/**
+	 * Sorts an array. Method is available to the client.
+	 * 
+	 * @param array
+	 *            A list of integers
+	 */
+	public static void mergeSort(int[] array) {
+		String stringifiedArray = Arrays.toString(array);
+		LOGGER.info("Array before sort: {}", stringifiedArray);
+		tempArray = new int[array.length];
```
Comment (ivorusev) [inline src/main/java/com/musala/edu/introduction/three/TaskThree.java]: What would happen with `tempArray` if I call the code async from different threads?

File: src/main/java/com/musala/edu/introduction/two/TaskTwo.java [hunk 4]
```diff
@@ -5,6 +5,12 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+/**
+ * The {@code TaskTwo} is a static class
+ * 
+ * @author georgi.kavalov
+ *
+ */
 public class TaskTwo {
```
Comment (ivorusev) [inline src/main/java/com/musala/edu/introduction/two/TaskTwo.java]: Again, some more meaningful name would be better.

File: src/test/java/com/musala/edu/introduction/three/TestTaskThree.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+import org.junit.Test;
+
+/**
+ * The {@code TestTaskThree} contains tests for the merge sort implementation
+ * 
+ * @author georgi.kavalov
+ *
+ */
+public class TestTaskThree {
```
Comment (ivorusev) [inline src/test/java/com/musala/edu/introduction/three/TestTaskThree.java]: Would be nice to add a test with null/empty array.

File: src/main/java/com/musala/edu/introduction/three/TaskThree.java (hunk 2 above)
Comment (georgikavalov) [reply src/main/java/com/musala/edu/introduction/three/TaskThree.java]: I thought it was obvious since it contains only one public method - mergeSort.
